Userland/Libraries/LibJS/AST.cpp

```diff
@@ -2020,7 +2020,7 @@ void RegExpLiteral::dump(int indent) const
 Value RegExpLiteral::execute(Interpreter& interpreter, GlobalObject& global_object) const
 {
     InterpreterNodeScope node_scope { interpreter, *this };
-    return regexp_create(global_object, js_string(interpreter.heap(), pattern()), js_string(interpreter.heap(), flags()));
+    return RegExpObject::create(global_object, move(m_regex), pattern(), flags());
```

Comment:
This `move(m_regex)` mean we can make just one regex object out of this. (Or at least I think this is the problem)
To reproduce:
```js
for (var i = 0; i < 2; i++)
{
    const re = /test/;
    console.log(re.test('te'));
    console.log(re.test('test'));
}
```

Which gives`false true false false`

Reply:
Damn, good catch. Maybe this should be a `RefPtr<Regex>`.

Reply:
Nope, `Regex` can't be ref-counted because it's used as an `NonnullOwnPtr` in `Userland/Libraries/LibRegex/C/Regex.cpp`:
```c++
 preg->re = make<Regex<PosixExtended>>(...);
```

Making `Regex` ref-counted causes the static assert in the `NonnullOwnPtr` constructor to fail:
```c++
static_assert(
    requires { requires typename T::AllowOwnPtr()(); } || !requires { requires !typename T::AllowOwnPtr()(); declval<T>().ref(); declval<T>().unref(); },
    "Use NonnullRefPtr<> for RefCounted types");
```

Reply:
Alrighty, so what I ended up doing is separating the parsing of the regex pattern and creating the `Regex<ECMA262>`  object. So the `RegExpLiteral` parses the pattern once, then creates a new `Regex<ECMA262>` with that pattern each time `RegExpLiteral::execute` is invoked.